lib/promise.js

```diff
@@ -1,33 +1,47 @@
 'use strict';
 
 Promise.each =  async function(arr, fn) {
-   for(const item of arr) await fn(item);
+    for(const item of arr) await fn(item);
 }
 
-Promise.prototype.return = func => {
-  return Promise.resolve(func);
+Promise.prototype.return = function(value) {
+  return this.then(function(){ return value; });
 }
 
 Promise.prototype.tap = function(fn) {
-    return this.then(function(value){
-         fn(value);
+    return this.then(async function(value){
+         await fn(value);
          return value;
     });
 };
 
-Promise.try = func => {
-  return new Promise((resolve, reject) => {
-  		resolve(func());
-  	});
-  }
+Promise.try = function(func) {
+    return new Promise(function(resolve, reject) {
+      try {
+        resolve(func());
+      }
+      catch (e) {
+        reject(e);
+      }
+    })
+}
 
-Promise.map = (fn, ctx) => {
+Promise.map = function(fn, ctx) {
   return Promise.resolve(function(val) {
     val = Array.isArray(val) ? val : [val]
     return Promise.all(val.map(fn, ctx))
   });
 }
 
+Promise.prototype.reflect = function() {
```

Comment:
Shouldn't this return an object which conforms with the `PromiseInspection` interface? Use at least in `_handleReflectedResult` and all model attribute validations.
http://bluebirdjs.com/docs/api/promiseinspection.html

Reply:
This just works with sequelize not following the spec.